src/js/dynload.ts

```diff
@@ -37,20 +62,35 @@ function createDynlibFS(
       }
     }
 
+    // If the path is absolute, we don't need to search for it.
+    if (Module.PATH.isAbs(path)) {
+      return path;
+    }
+
+    // Step 1) Try to find the library in the search directories
     for (const dir of _searchDirs) {
       const fullPath = Module.PATH.join2(dir, path);
+
       if (Module.FS.findObject(fullPath) !== null) {
         return fullPath;
       }
     }
+
+    // Step 2) try to find the library by searching child directories of the library directory
+    //         (This should not be necessary in most cases, but some libraries have dependencies in the child directories)
```

Comment:
This is odd. Does normal linux have logic like this? Or does it have some different behavior with `LD_LIBRARY_PATH` or something?

Reply:
Locating the path of the shared library is handled two ways in linux.

1. Using `LD_LIBRARY_PATH`
2. Using `rpath`

In emscripten, `LD_LIBRARY_PATH` is used [in _dlopen](https://github.com/emscripten-core/emscripten/blob/82b0d740924f589b320c624191497e1ab7857edc/system/lib/libc/dynlink.c#L518-L527), but we are not using it.

`rpath` is not supported in emscripten. I made a proposal to add it before (https://github.com/pyodide/pyodide/issues/3854), I guess I need to open a same proposal in emscripten too.

Reply:
Can we add a `// TODO: add rpath to Emscripten dsos and remove this logic`?